src/threadpool.c

```diff
@@ -50,10 +54,16 @@ static void uv__cancelled(struct uv__work* w) {
  * never holds the global mutex and the loop-local mutex at the same time.
  */
 static void worker(void* arg) {
+  struct uv__thread_start* ts;
   struct uv__work* w;
   QUEUE* q;
 
-  (void) arg;
+  ts = arg;
+  uv_sem_wait(&ts->sem);
```

Comment:
Don't wait, just post. The init thread can wait nthreads times, once for each post.